adapters/backend/v1/pulsar.go

```diff
@@ -31,6 +31,8 @@ const (
 // * Pulsar Message Reader  *//
 // ******************************
 
+var reconciliationMessageMutex sync.Mutex
```

Comment:
can you move the mutex to the PulsarMessageReader? I tend to avoid global locks when it's not needed